src/njs_parser.c

```diff
@@ -4630,8 +4650,9 @@ njs_parser_statement_wo_node(njs_parser_t *parser, njs_lexer_token_t *token,
         return njs_parser_variable_statement(parser, token, current);
 
     case NJS_TOKEN_SEMICOLON:
-        njs_lexer_consume_token(parser->lexer, 1);
-        return njs_parser_stack_pop(parser);
+        /* Insert empty block statement. */
+        njs_parser_next(parser, njs_parser_empty_block_statement);
```

Comment:
I do not think we should introduce a block statement for any empty statement, it seems not needed.

Reply:
The idea localize changes in njs_parser_labelled_statement_after(), I'll accept it as it prevent more of already existent code. While still think better use general approach, when parser's business is just producing regular tree without intervention into it (like removing empty statement nodes, and so on). It make parser code more simple and clear.